src/MSALWrapper/GlobalLock.cs

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.Authentication.MSALWrapper
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Text;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    /// <summary>
+    /// GlobalLock is used for preventing code from entering critical sections in different processes at the same time.
+    /// </summary>
+    public class GlobalLock : IDisposable
+    {
+        private Mutex mutex;
+        private bool disposedValue;
+        private bool lockAcquired;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="GlobalLock"/> class.
+        /// </summary>
+        /// <param name="lockName">Unique lock ID.</param>
+        /// <exception cref="ArgumentNullException">The <see cref="ArgumentNullException"/>.</exception>
+        public GlobalLock(string lockName)
+        {
+            if (lockName == null)
+            {
+                throw new ArgumentNullException($"{nameof(lockName)} must not be null");
+            }
+
+            this.mutex = new Mutex(false, lockName);
+            try
+            {
+                this.mutex.WaitOne();
+                this.lockAcquired = true;
+            }
+            catch (AbandonedMutexException)
+            {
+                // lock eventually acquired
+                this.lockAcquired = true;
+            }
+        }
+
+        /// <summary>
+        /// Dispose method.
+        /// </summary>
+        /// <param name="disposing">disposing.</param>
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!this.disposedValue)

```

Comment:
I _think_ I understand what `disposing` is for, but what is `disposedValue` for?